raiden/tests/integration/api/test_pythonapi.py

```diff
@@ -23,6 +23,58 @@ def test_token_addresses(raiden_network, token_addresses):
     )
 
 
+@pytest.mark.parametrize("number_of_nodes", [2])
+@pytest.mark.parametrize("number_of_tokens", [1])
+@pytest.mark.parametrize("channels_per_node", [0])
+def test_channel_open_token_network_not_registered_in_confirmed_block(
+    raiden_network, token_addresses
+):
+    """
+    Test that opening a channel via the API provides the confirmed block and not
+    the latest block. The discrepancy there lead to potential timing issues where
+    the token network was deployed for the state in the "latest" block but not yet
+    in the confirmed state and a BadFunctionCallOutput exception was thrown from web3.
+
+    Regression test for 4470
+    """
+    app0, app1 = raiden_network
+    token_address = token_addresses[0]
+
+    # Find block where the token network was deployed
```

Comment:
Question: Anybody knows of a better way to find a block a contract was deployed?